server/datastore/mysql/hosts.go

```diff
@@ -1926,68 +1943,83 @@ WHERE
 // Aggregated stats for os versions are stored by team id with 0 representing the global case
 // If existing team has no hosts, we explicity set the json value as an empty array
 func (ds *Datastore) UpdateOSVersions(ctx context.Context) error {
+	// TODO: counts will be lagging until offline hosts report back with detail query
+
 	sql := `
 INSERT INTO aggregated_stats (id, type, json_value)
 SELECT
-  team_id id,
-  'os_versions' type,
-  COALESCE(
-    CONCAT('[', GROUP_CONCAT(
-      JSON_OBJECT(
-        'hosts_count', hosts_count,
-        'name', name,
-        'platform', platform
-      )
-    ), ']'),
-    JSON_ARRAY()
-  ) json_value
-FROM
-  (
-    SELECT
-      COUNT(*) hosts_count,
-      h.os_version name,
-      h.platform,
-      0 team_id
-    FROM
-      hosts h
-    GROUP BY
-      h.os_version,
-      h.platform
-    UNION
-    SELECT
-      COUNT(*) hosts_count,
-      h.os_version name,
-      h.platform,
-      h.team_id
-    FROM
-      hosts h
-      JOIN teams t ON t.id = h.team_id
-    GROUP BY
-      h.os_version,
-      h.platform,
-      h.team_id
-  ) as team_os_versions
+	team_id id,
+	'os_versions' type,
+	COALESCE(
+		CONCAT('[', GROUP_CONCAT(
+			JSON_OBJECT(
+				'hosts_count', hosts_count, 
+				'name', name, 
+				'name_only', name_only, 
+				'version', version, 
+				'os_id', os_id, 
+				'platform', platform
+			)
+		), ']'), 
+		JSON_ARRAY()
+	) json_value
+FROM (
+	SELECT
+		team_id,
+		os_id,
+		hosts_count,
+		CONCAT(name, ' ', version)
+		name,
+		name name_only,
+		version,
```

Comment:
For documenting purposes, prefix these fields with `os.` (these are coming from the `LEFT JOIN operating_systems os ON os.id = os_id`)